connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/controller/InboundInstancesRestController.java

```diff
@@ -36,17 +47,49 @@
 @RequestMapping("/inbound-instances")
 public class InboundInstancesRestController {
 
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(InboundInstancesRestController.class);
+
   private final InboundExecutableRegistry executableRegistry;
 
   private final ConnectorDataMapper connectorDataMapper = new ConnectorDataMapper();
 
-  public InboundInstancesRestController(InboundExecutableRegistry executableRegistry) {
+  private final InstanceForwardingService instanceForwardingService;
+
+  private static final String HOSTNAME = System.getenv("HOSTNAME");
+
+  public InboundInstancesRestController(
+      InboundExecutableRegistry executableRegistry,
+      @Autowired(required = false) InstanceForwardingService instanceForwardingService) {
     this.executableRegistry = executableRegistry;
+    this.instanceForwardingService = instanceForwardingService;
+  }
+
+  /**
+   * This method is used to forward the request to the instances or local service depending on the
+   * configuration.
+   *
+   * @see io.camunda.connector.runtime.instances.InstanceForwardingConfiguration
+   */
+  private <T> T forwardAndReduceToInstancesOrLocal(
+      HttpServletRequest request, String forwardedFor, Supplier<T> supplier) {
+    if (instanceForwardingService != null && StringUtils.isBlank(forwardedFor)) {
+      LOGGER.debug(
+          "Forwarding request to instances: {}",
+          request.getRequestURL().toString() + "?" + request.getQueryString());
+      return instanceForwardingService.forwardAndReduce(request, new TypeReference<>() {});
+    }
```

Comment:
As written above: I would recommend to move this to an `InboundInstanceService` implementation that is able to aggregate like a `AggregatingInboundInstanceService` implementation. In addition to that we could have a `LocalInboundInstanceService` that is just using the registry.